src/goto-symex/dynamic_allocation.cpp

```diff
@@ -84,8 +84,7 @@ void goto_symext::default_replace_dynamic_allocation(expr2tc &expr)
     expr2tc alloc_arr_2;
     migrate_expr(symbol_expr(*ns.lookup(valid_ptr_arr_name)), alloc_arr_2);
 
-    index2tc index_expr(get_bool_type(), alloc_arr_2, obj_expr);
-    expr = not2tc(index_expr);
+    expr = index2tc(get_bool_type(), alloc_arr_2, obj_expr);
```

Comment:
This modification does not wrap `index2tc` in a `not2tc`. It's breaking the following test cases:
```
  356 - regression/esbmc/00_memory_leak_06 (Failed)
  357 - regression/esbmc/00_memory_leak_07 (Failed)
  363 - regression/esbmc/00_memory_leak_13 (Failed)
  366 - regression/esbmc/00_memory_leak_16 (Failed)
  525 - regression/esbmc/github_212 (Failed)
  561 - regression/esbmc/github_330 (Failed)
  562 - regression/esbmc/github_330_1 (Failed)
  568 - regression/esbmc/github_330_7 (Failed)
  598 - regression/esbmc/github_387_true (Failed)
  982 - regression/cbmc/01_cbmc_Pointer32 (Failed)
```

Reply:
@fbrausse could you give us a hand here?

Reply:
commit 3c9f3a02b3fed7ca11a73191a1824b1c24155ba7 fixed these failues.